TOKEN_GUIDE.md

```diff
@@ -0,0 +1,132 @@
+---
+title: Build a Token Generator with Astro
+description: Learn how you can generate tokens using SSR frameworks like Astro.
+---
+
+I love building websites with Astro. It's my favorite web framework, and I don't want to use any other one. However, I must stress test it and ensure Astro is flexible enough to fully commit.
+
+Astro is created to be the perfect use case for a content-heavy website. The opposite of that is a backend, and in this guide, we will build a backend token generator to be used for Agora video calls.
+
+Spoilers. My love for Astro has grown deeper. 
+
+![Token Flow](/assets/token-flow.png)
+
+## Prerequisites
+1. NodeJS and Astro installed.
+2. A developer account with [Agora](https://console.agora.io/).
+
+## Project Setup
+To create a new Astro project, run `npm create astro@latest`. Follow all the default options during the setup process, except use an empty template. This way, our project will be free of distractions. Then, we must install the Agora Token package to generate our tokens. You can do that using `npm i agora-token`.
+
+You must also create a `.env` and add `APP_ID` and `APP_CERTIFICATE` values, which you can find within your Agora Console.
```

Comment:
You should explain a bit more about the .env file here. To give the user the reason "why" these values should be in the .env file. 

for example, is this data sensitive? or do we use the .env to switch between dev and production environments?

Reply:
Updated to 

> You must also create a `.env` and add `APP_ID` and `APP_CERTIFICATE` values, which you can find within your Agora Console. This information is used to create tokens and join video calls, so it's best not to show these values publicly. When you create an Astro project, it will create a `.gitignore` file for you. This file will have `.env` inside, meaning the `.env` file will not be pushed to GitHub or other git services.

Reply:
1. APP_CERTIFICATE is a sensitive key, and users should be warned to be careful and protect it from public view.
2. The •gitignore is a great touch but the way it's phrased makes it tough to follow without re-reading a few times.

Not saying you have to phrase it this way, just an example of how I would explain it, to give extra context as to why its important to use an env file to store sensitive information like API credentials:
> In this project we're going to use an `APP_CERTIFICATE` to generate tokens. This is our "secret" key and should be shielded from public view. We'll create a `.env` file to store our sensitive data. The `. env `file is industry
standard; it's already listed in the `.gitignore` file generated when we created our Astro project.
In the env, add the project `APP_ID` and `APP_CERTIFICATE`; you can find these in your Agora Console.

Reply:
Reworded to this: 

> To generate a token, we need to use our Agora App ID and App Certificate. The App Certificate is our "secret" key and should be hidden from public view. We will use a `.env` file to store our sensitive data. When we generated our Astro project, the `.env` was already listed in the `.gitignore`, which means it will not be pushed to GitHub. In the `.env`, add the project `APP_ID` and `APP_CERTIFICATE`, which you can find in your Agora Console.